src/Sulu/Bundle/RouteBundle/Manager/RouteManager.php

```diff
@@ -98,24 +87,60 @@ public function update(RoutableInterface $entity, $path = null, $resolveConflict
             return $entity->getRoute();
         }
 
-        $historyRoute = $entity->getRoute()->setHistory(true)->setTarget($route);
-        $route->addHistory($historyRoute);
+        $route = $this->handleHistoryRoutes($entity->getRoute(), $route);
+
+        $entity->setRoute($route);
+
+        return $route;
+    }
+
+    public function createOrUpdateByAttributes(string $entityClass, string $id, string $locale, string $path): RouteInterface
```

Comment:
What if we create a `RouteFactory::create` service and the above function use it also?

Reply:
why a service? a method inside this class would also be enough i think :)

Reply:
Single responsibility principe 😄